rs/types/management_canister_types/src/lib.rs

```diff
@@ -2251,6 +2251,131 @@ impl FromStr for SchnorrKeyId {
+    Bls12_381_G2,
+}
+
+impl From<&VetKdCurve> for pb_registry_crypto::VetKdCurve {
+    fn from(item: &VetKdCurve) -> Self {
+        match item {
+            VetKdCurve::Bls12_381_G2 => pb_registry_crypto::VetKdCurve::VetkdCurveBls12381G2,
+        }
+    }
+}
+
+impl TryFrom<pb_registry_crypto::VetKdCurve> for VetKdCurve {
+    type Error = ProxyDecodeError;
+
+    fn try_from(item: pb_registry_crypto::VetKdCurve) -> Result<Self, Self::Error> {
+        match item {
+            pb_registry_crypto::VetKdCurve::VetkdCurveBls12381G2 => Ok(VetKdCurve::Bls12_381_G2),
+            pb_registry_crypto::VetKdCurve::VetkdCurveUnspecified => {
+                Err(ProxyDecodeError::ValueOutOfRange {
+                    typ: "VetKdCurve",
+                    err: format!("Unable to convert {:?} to a VetKdCurve", item),
+                })
+            }
+        }
+    }
+}
+
+impl std::fmt::Display for VetKdCurve {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl FromStr for VetKdCurve {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "bls12_381_g2" => Ok(Self::Bls12_381_G2),
+            _ => Err(format!("{} is not a recognized vetKD curve", s)),
+        }
+    }
+}
+
+/// Unique identifier for a key that can be used for threshold key derivation
+/// (vetKD). The name is just an identifier, but it may be used to convey
+/// some information about the key (e.g. that the key is meant to be used for
+/// testing purposes).
+/// ```text
+/// (record { curve: vetkd_curve; name: text})
+/// ```
+#[derive(
+    Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, CandidType, Deserialize, Serialize,
+)]
+pub struct VetKdKeyId {
+    pub curve: VetKdCurve,
+    pub name: String,
+}
+
+impl From<&VetKdKeyId> for pb_registry_crypto::VetKdKeyId {
+    fn from(item: &VetKdKeyId) -> Self {
+        Self {
+            curve: pb_registry_crypto::VetKdCurve::from(&item.curve) as i32,
+            name: item.name.clone(),
+        }
+    }
+}
+
+impl TryFrom<pb_registry_crypto::VetKdKeyId> for VetKdKeyId {
+    type Error = ProxyDecodeError;
+    fn try_from(item: pb_registry_crypto::VetKdKeyId) -> Result<Self, Self::Error> {
+        Ok(Self {
+            curve: VetKdCurve::try_from(
+                pb_registry_crypto::VetKdCurve::try_from(item.curve).map_err(|_| {
+                    ProxyDecodeError::ValueOutOfRange {
+                        typ: "VetKdKeyId",
+                        err: format!("Unable to convert {} to a VetKdCurve", item.curve),
+                    }
+                })?,
+            )?,
+            name: item.name,
+        })
+    }
+}
+
+impl std::fmt::Display for VetKdKeyId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}:{}", self.curve, self.name)
+    }
+}
+
+impl FromStr for VetKdKeyId {
```

Comment:
FYI there are also some unit tests for the (de)serialization of the existing types at the bottom of this file

Reply:
Thanks! Added the respective tests vor vetKD: https://github.com/dfinity/ic/pull/2108/commits/fd4a1c5cb545d0d9717b6efdb333bee6eb435b51